.ci/scripts/test.sh

```diff
@@ -119,37 +116,100 @@ if [ "${interface}" == "daal/java" ]; then
 fi
 
 for link_mode in ${link_modes}; do
-    # Release Examples testing
     if [ "${link_mode}" == "static" ]; then
+        lib_ext="a"
         l="lib"
     elif [ "${link_mode}" == "dynamic" ]; then
         if [ "${OS}" == "lnx" ]; then
+            lib_ext="so"
             l="so"
         else
+            lib_ext="dylib"
             l="dylib"
         fi
     fi
-    build_command="make ${make_op} ${l}${full_arch} mode=build compiler=${compiler}"
-    echo "Building ${TEST_KIND} ${build_command}"
-    (${build_command})
-    err=$?
-    if [ ${err} -ne 0 ]; then
-        echo -e "$(date +'%H:%M:%S') BUILD FAILED\t\t${link_mode}"
-        TESTING_RETURN=${err}
-        continue
-    else
-        echo -e "$(date +'%H:%M:%S') BUILD COMPLETED\t\t${link_mode}"
-    fi
-    run_command="make ${l}${full_arch} mode=run compiler=${compiler}"
-    echo "Running ${TEST_KIND} ${run_command}"
-    (${run_command})
-    err=$?
-    if [ ${err} -ne 0 ]; then
-        echo -e "$(date +'%H:%M:%S') RUN FAILED\t\t${link_mode} with errno ${err}"
-        TESTING_RETURN=${err}
-        continue
+    if [ "$build_system" == "cmake" ]; then
+        if [[ ${compiler} == gnu ]]; then
+            export CC=gcc
+            export CXX=g++
+        elif [[ ${compiler} == clang ]]; then
+            export CC=clang
+            export CXX=clang++
+        elif [[ ${compiler} == icx ]]; then
+            export CC=icx
+            export CXX=icpx
+        fi
+        echo "============== Configuration: =============="
+        echo Compiler:  ${compiler}
+        echo Link mode: ${link_mode}
+        echo CMAKE_C_COMPILE: ${CC}
+        echo CMAKE_CXX_COMPILER: ${CXX}
```

Comment:
These echos might be misleading because CMake may use compiler not from CC/CXX in some cases

Reply:
Do you have better examples how to get relevant compilers? 
Agree on naming - would change to just CC and CXX

Reply:
cmake outputs used compilers itself:
```
-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
```
so, output of preset CC and CXX will be enough